Mathlib/Algebra/Algebra/Basic.lean

```diff
@@ -74,41 +74,46 @@ theorem _root_.ULift.down_algebraMap (r : R) : (algebraMap R (ULift A) r).down =
 
 end ULift
 
+section SubsemiringAlgebra
+
+variable {C : Type*} [SetLike C R] [SubsemiringClass C R]
+
 /-- Algebra over a subsemiring. This builds upon `Subsemiring.module`. -/
-instance ofSubsemiring (S : Subsemiring R) : Algebra S A where
-  algebraMap := (algebraMap R A).comp S.subtype
+instance (priority := 900) ofSubsemiring (S : C) : Algebra S A where
+  algebraMap := (algebraMap R A).comp (Subsemiring.subtype <| .ofClass S)
   commutes' r x := Algebra.commutes (r : R) x
   smul_def' r x := Algebra.smul_def (r : R) x
```

Comment:
The main reason of the slowdown appeared to be `Algebra ↥v.valuationSubring ↥(integralClosure (↥v.valuationSubring) L)` getting this instance, which isn't trivially defeq to the instance from [Subalgebra.algebra](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Algebra/Subalgebra/Basic.html#Subalgebra.algebra), and Lean spent a lot of time checking the defeq. Lowering the priority solves this.